flink/v1.17/flink/src/main/java/org/apache/iceberg/flink/sink/RowDataTaskWriterFactory.java

```diff
@@ -117,6 +145,8 @@ public TaskWriter<RowData> create() {
         outputFileFactory,
         "The outputFileFactory shouldn't be null if we have invoked the initialize().");
 
+    FileIO io = tableSupplier.get().io();
```

Comment:
Why is this needed?

Reply:
Mainly so it could be reused in the different writer constructors. I went ahead and got rid of it.